src/services/forecast.ts

```diff
@@ -44,13 +59,17 @@ export class Forecast {
   private async calculateRating(beaches: Beach[]): Promise<BeachForecast[]> {
     const pointsWithCorrectSources: BeachForecast[] = [];
     logger.info(`Preparing the forecast for ${beaches.length} beaches`);
-    for (const beach of beaches) {
-      const rating = new this.RatingService(beach);
-      //TODO someone to make this call in parallel
-      const points = await this.stormGlass.fetchPoints(beach.lat, beach.lng);
-      const enrichedBeachData = this.enrichBeachData(points, beach, rating);
-      pointsWithCorrectSources.push(...enrichedBeachData);
-    }
+    const points: ForecastPoint[][] = await Promise.all(
+      beaches.map((beach) => this.stormGlass.fetchPoints(beach.lat, beach.lng))
+    );
+
+    points.forEach((point: ForecastPoint[], index: number) => {
+      const ratingService = new this.RatingService(beaches[index]);
```

Comment:
How are you going to ensure that the beach you get here `this.RatingService(beaches[index]` is the one that the `point` belongs to? From what I see on the code there's no way to ensure that they will be in the right order.

Also, keep in mind that there will be more points than beaches which would result in `beaches[index]` returning `undefined`, typescript should complain about it actually xD.

Reply:
Here I make the API calls in parallel so the function ** promise.all ** ensures that the order is the same as the beaches passed by parameter, as the points will be grouped by beach and the order will be the same regardless of the response time of each API called.

Reply:
niceee, yeah you are right! I should've read the spec, the `promise.all` keep an index marker to ensure that the output is the same. Awesome!

Reply:
About my second point, I probably got confused by the names, `points` is actually `an array` of `an array of points`, let's rename it to just `response` instead of `points`, it's going to be easier to understand?.
And what do you think of using this opportunity to get rid of the `pointsWithCorrectSources`? was thinking about something like this:

```
    return points.flatMap((point, index) => {
      const ratingService = new this.RatingService(beaches[index]);
      return this.enrichBeachData(point, beaches[index], ratingService);
    });
```
Also `points` would be renamed to `response` and the `point` should be called `points` because it's an array.
let me know what you think, I think it's way more clear. Also it's preferable to use iterators (map, for of/in) over common loops like forEach (https://stackoverflow.com/a/50844178/4043593)

Reply:
great! the code was much cleaner and readable, thank you for the tips.